python/modules/IcePy/Types.cpp

```diff
@@ -1476,18 +1469,17 @@ IcePy::SequenceInfo::marshal(
             Py_ssize_t sz = 0;
             if (p != Py_None)
             {
-                const void* buf = 0;
-                if (PyObject_AsReadBuffer(p, &buf, &sz) == 0)
+                Py_buffer pybuf;
+                if (pi && PyObject_GetBuffer(p, &pybuf, PyBUF_SIMPLE | PyBUF_FORMAT) == 0)
                 {
-                    if (pi->kind == PrimitiveInfo::KindString)
-                    {
-                        PyErr_Format(PyExc_ValueError, "expected sequence value");
-                        throw AbortMarshaling();
-                    }
+                    // Strings are handled as variable length types above.
+                    assert(pi->kind != PrimitiveInfo::KindString);
+                    sz = pybuf.len;
+                    PyBuffer_Release(&pybuf);
```

Comment:
each successful call to ` PyObject_GetBuffer` must be paired with a call to PyBuffer_Release to avoid leaks.